app/code/Magento/Deploy/Model/Filesystem.php

```diff
@@ -279,10 +295,30 @@ protected function compile(OutputInterface $output)
      * @param array $directoryCodeList
      * @return void
      * @throws \Magento\Framework\Exception\FileSystemException
+     * @deprecated
+     * @see cleanupFilesystemDirectories
      */
     public function cleanupFilesystem($directoryCodeList)
     {
-        $excludePatterns = ['#.htaccess#', '#deployed_version.txt#'];
+        $this->cleanupFilesystemDirectories($directoryCodeList, false);
+    }
+
+    /**
+     * Deletes specified directories by code
+     *
+     * @param array $directoryCodeList
+     * @param bool $resetStaticVersion
+     * @return void
+     * @throws \Magento\Framework\Exception\FileSystemException
+     */
+    public function cleanupFilesystemDirectories($directoryCodeList, $resetStaticVersion = false)
```

Comment:
Unfortunately, we are not allowed to add new public methods according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)